pyteal/compiler/sourcemap.py

```diff
@@ -857,6 +866,16 @@ def _validate_build(self):
                     f"TealMapItem's don't match R3SourceMap.file_lines at index {i}. ('{tmi_line}' v. '{target_line}')"
                 )
 
+        for tmi in self._cached_tmis:
+            if not tmi.is_sentinel:
+                continue
+            print(
+                f"""-----------------
+WARNING: Source mapping is unknown for the following:
+{tmi!r}
+"""
```

Comment:
if I understood correctly, this warning will happen also in normal compilation with sourcemap (and correctly set feature gate), right?

Reply:
I would be very surprised if this happens when feature gates are imported _prior _to pyteal and set correctly. If this occurs it would be a bug.

As a sanity check I just ran locally and got the following:
```
❯ cd examples/application
❯ python sourcemap.py
SUCCESS!!! Please check out teal/sourcemap.teal and teal/sourcemap_annotated.teal
```

The logic here is:

1. Insert a **sentinel** `TealMapItem` when no _best frame_ is found for a `TealComponent`
2. In `_validate_build()` look for such a sentinel. Only in the case that it is found print the error message.

Reply:
Ran another sanity check, this time on [the AlgoBank example](https://github.com/algorand/pyteal/blob/3a7bfa2a1aa91b2713b5f96275b847c38b4313bd/examples/application/abi/algobank.py#L143) -but brough the `... import FeatureGate ...` stuff to the top.

Reply:
oh my bad, I was confused! The logic got me into the wrong point is that, I wrongly assume the case that `self._best_frame` would contain `None` before this PR, but *only under wrongly set FeatureGate* would cause such issue, and thus we introduce the sentinel_frame to trigger the warning!

Now I understood... resolving